src/platform_impl/macos/app_delegate.rs

```diff
@@ -125,6 +130,18 @@ extern "C" fn application_open_urls(_: &Object, _: Sel, _: id, urls: id) -> () {
   trace!("Completed `application:openURLs:`");
 }
 
+extern "C" fn application_should_handle_reopen(
+  _: &Object,
+  _: Sel,
+  _: id,
+  has_visible_windows: BOOL,
+) -> BOOL {
+  trace!("Triggered `applicationShouldHandleReopen`");
+  AppState::reopen(has_visible_windows != NO);
+  trace!("Completed `applicationShouldHandleReopen`");
+  has_visible_windows
```

Comment:
I'm not sure if using has_visible_windows as the return type is a good idea. 🤔 

It's required to make event_loop.hide_application() to keep working out of the box but it may still cause incorrect behavior somewhere (or even with hide_application if the user doesn't want the default show behavior).

I'm not sure how we could best expose this to tauri consumers etc so maybeee just leave it like this for now (but adding a comment acknowledging that the logic is technically incorrect) and see where this will get us 🤔 

What do you think @pewsheen ?

Reply:
some context: https://github.com/electron/electron/blob/main/shell/browser/mac/electron_application_delegate.mm#L108-L113

(I tried some simple/cheap method to pass a thing to the user, but all of them failed :(
(is passing a wrapper struct around a shared reference to mutex/atomic and expose a method to modify it acceptable?

Reply:
> some context: https://github.com/electron/electron/blob/main/shell/browser/mac/electron_application_delegate.mm#L108-L113

Interesting that electron does it too, then i guess it won't be too bad and just validates my statement above to just try this as is :)

> is passing a wrapper struct around a shared reference to mutex/atomic and expose a method to modify it acceptable?

Sometimes yes. In wry we also do this in some places with a static OnceCell or Lazy. In some places also use objc ffi to set a function on the object/delegate for later use but i personally prefer the full rust solution (the static vars) over that shitshow.

Reply:
https://github.com/BillGoldenWater/tao/compare/dev...BillGoldenWater:tao:feat/reopen_ret_test

implemented a test for expose this to user, could you take a look?

Reply:
Basically if `hasVisibleWindows = true` and you return `true` from `applicationShouldHandleReopen()` then the system calls [`applicationOpenUntitledFile()`](https://developer.apple.com/documentation/appkit/nsapplicationdelegate/1428491-applicationopenuntitledfile) call. I think electron got it wrong.  But since we don't handle `applicationOpenUntitledFile()` then it really doesn't matter all that much.

Reply:
I think we could go this way first and see if anyone has the need to control the return value. If so, we can add something like https://github.com/tauri-apps/wry/pull/1196 to accept a handler from caller.

Reply:
Yeah i agree, maybe by then winit also has more ideas for their platform specific events 😂 

If electron did this for 10 years i think it's good enough for us too x) At least i didn't see any complains about it in electron-land.

Reply:
cc @amrbashir because 🕺

Reply:
this is fine for me as is for now, let's ship this and worry about other stuff later when requestred